core/base/model/AliceSkill.py

```diff
@@ -137,26 +177,26 @@ def loadWidgets(self):
 				)
 
 
+	def getWidgetInstance(self, widgetName: str) -> Optional[Widget]:
+		return self._widgets.get(widgetName)
+
+
 	def getUtterancesByIntent(self, intent: Intent, forceLowerCase: bool = True, cleanSlots: bool = False) -> list:
-		utterances = list()
+		lang = self.LanguageManager.activeLanguage
+		if not lang in self._intentsDefinitions:
+			return list()
 
 		if isinstance(intent, tuple):
 			check = intent[0].justAction
 		elif isinstance(intent, Intent):
 			check = intent.justAction
 		else:
-			check = intent.split('/')[-1].split(':')[-1]
-
-		for dtIntentName, dtModuleName in self.SamkillaManager.dtIntentNameSkillMatching.items():
-			if dtIntentName == check and dtModuleName == self.name:
+			check = str(intent).split('/')[-1].split(':')[-1]
 
-				for utterance in self.SamkillaManager.dtIntentsModulesValues[dtIntentName]['utterances']:
-					if cleanSlots:
-						utterance = re.sub(self._utteranceSlotCleaner, '\\1', utterance)
+		if not check in self._intentsDefinitions[lang]:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [test for membership should be 'not in'](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4699682)